### ISSUE TITLE
Make protocol and non-protocol contract seperation clearer

### ISSUE THREAD

**Issue by adamdossa** [PolymathNetwork/polymath-core-deprecated#137]
e.g. STOContract is not part of the protocol. This should also allow us to add non-core protocol pieces (e.g. sample STO contracts) in a clean way.

**Comment by pabloruiz55** [comment]
Hey @adamdossa, does this mean also handling the separation of concerns between ST and STO?

**Comment by pabloruiz55** [comment]
We need to change the way STOs and offering templates work. I'd rework it like this: 

Have contributors build Factory contracts for STOs. These Factory contracts would be published by us and/or the community and people wanting to implement them would call a method on the Factory contract that deploys the corresponding STO.

For example, a Standard STO Factory contract would have a method to deploy an instance of an Standard STO contract that receives a couple of parameters  (start/end time, rate, accepts POLY, ST address). 

The Factory contract would define an address for the beneficiary to receive POLY paid by issuers deploying STOs from it. 

The Factory would keep track of the usage metrics of itself (contracts deployed by it, accumulated POLY raised, etc, etc).

There could be (or not) a Factory Registrar contract that keeps track of Factories deployed.

Issuers wanting to use a certain STO would only have to deploy the ST, choose a Factory that matches the properties of the STO they want and deploy it, and associate the token to that STO.

The intention of the above is to eliminate the duplication we have with template offerings and STOs how they work together and make the process of STO creation more streamlined.
This would also be the first step towards decoupling STs and STOs.

-----

The above looks like a good candidate to make it a PIP (PIP-2).

-----

@adamdossa @Everhusk @ChainSafe  thoughts?

**Comment by adamdossa** [comment]
Hi @pabloruiz55 - I took a slightly different route, although I can see pros / cons vs. your approach above. I changed Compliance / SecurityToken to accept OfferingFactory proposals (rather than Offering contracts) and then when you start the offering it generates a new instance from the factory for your ST. The reputation is then tracked at the Offering Factory level.

Take a look at the PR and let me know your thoughts re. the proposal above.
